app/component/itinerary/Navigator.js

```diff
@@ -51,9 +51,13 @@ function Navigator({ itinerary, focusToLeg, setNavigation }, context) {
     if (newLeg && newLeg !== currentLeg) {
       setCurrentLeg(newLeg);
       focusToLeg(newLeg, false);
+      // set nextLeg
+      const next = itinerary.legs.find(
+        leg => legTime(leg.start) > legTime(newLeg.start),
+      );
+      setNextLeg(next);
     }
```

Comment:
I wonder if next leg could simply be searched just before rendering NaviLeg. It does not have to be  in component state.